src/TwoOutOfThreeRowVisitor.cpp

```diff
@@ -50,15 +52,14 @@ bool TwoOutOfThreeRowVisitor::Visit(Grid &ioGrid) const{
 
 
                                         // find the column where it should be
-                                        
                                         presentInTriplet = vector<bool>(3, false);
  
-                                        presentInTriplet[0] = ioGrid.getRegion(3*((main_row+1)%3) + (i+0)%3).leftColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (i+0)%3).leftColumn().isValuePresent(value); 
-                                        presentInTriplet[1] = ioGrid.getRegion(3*((main_row+1)%3) + (i+0)%3).middleColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (i+0)%3).middleColumn().isValuePresent(value); 
-                                        presentInTriplet[2] = ioGrid.getRegion(3*((main_row+1)%3) + (i+0)%3).rightColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (i+0)%3).rightColumn().isValuePresent(value); 
+                                        // is there this value already in this column?
+                                        presentInTriplet[0] = ioGrid.getRegion(3*((main_row+1)%3) + (main_col+0)%3).leftColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (main_col+0)%3).leftColumn().isValuePresent(value); 
+                                        presentInTriplet[1] = ioGrid.getRegion(3*((main_row+1)%3) + (main_col+0)%3).middleColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (main_col+0)%3).middleColumn().isValuePresent(value); 
+                                        presentInTriplet[2] = ioGrid.getRegion(3*((main_row+1)%3) + (main_col+0)%3).rightColumn().isValuePresent(value) || ioGrid.getRegion(3*((main_row+2)%3) + (main_col+0)%3).rightColumn().isValuePresent(value); 
 
-                                        // Cannot be place in cells that are not empty
-                                        
+                                        // get the possible column in the working region                                        
                                         if(line == 0) {
```

Comment:
As said in the last PR, please use https://github.com/albanD/sudoku-solver/blob/master/src/RegionHolder.cpp#L54
instead of this